cranelift/codegen/src/machinst/lower.rs

```diff
@@ -771,6 +771,19 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
                     debug_assert_eq!(regs.len(), dsts.len());
                     for (dst, temp) in dsts.regs().iter().zip(regs.regs().iter()) {
                         self.set_vreg_alias(*dst, *temp);
+
+                        // If there was any PCC fact about the
+                        // original VReg, move it to the aliased reg
+                        // instead. Lookup goes through the alias, but
+                        // we want to preserve whatever was stated
+                        // about the vreg before its producer was
+                        // lowered.
+                        if let Some(fact) =
+                            self.vregs.take_fact(dst.to_virtual_reg().unwrap().into())
+                        {
+                            self.vregs
+                                .set_fact(temp.to_virtual_reg().unwrap().into(), fact);
+                        }
```

Comment:
Do we know we won't ever look up facts for the aliased vreg anymore? That is, is it really safe to `take` rather than `clone`?

Reply:
The lookup follows the alias, so querying the alias will get the fact as well -- we're moving it here so if the aliaser had a fact and the aliasee didn't, the fact gets preserved (that's the common case with the way ISLE lowering works -- we alias the original reg to the new temp the rule returned).